test/websocket/issue-3546.js

```diff
@@ -0,0 +1,16 @@
+'use strict'
+
+const { test } = require('node:test')
+const { WebSocket } = require('../..')
+const { tspl } = require('@matteo.collina/tspl')
+
+test('close is fired on failed connection', async (t) => {
+  const { completed, deepStrictEqual } = tspl(t, { plan: 1 })
+  const ws = new WebSocket('ws://localhost:1')
+
+  ws.addEventListener('close', () => {
+    deepStrictEqual(ws.readyState, WebSocket.CLOSED)
+  })
+
+  await completed
+})
```

Comment:
imho check the right order too

```suggestion
test('first error than close event is fired on failed connection', async (t) => {
  const { completed, strictEqual } = tspl(t, { plan: 4 })
  const ws = new WebSocket('ws://localhost:1')

  let orderOfEvent = 0

  ws.addEventListener('error', () => {
    strictEqual(orderOfEvent++, 0)
    strictEqual(ws.readyState, WebSocket.CLOSED)
  })

  ws.addEventListener('close', () => {
    strictEqual(orderOfEvent++, 1)
    strictEqual(ws.readyState, WebSocket.CLOSED)
  })

  await completed
})
```